azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Network.java

```diff
@@ -0,0 +1,208 @@
+         * @return the next stage of the virtual network update
+         */
+        Update withoutSubnet(String name);
+    }
+
+    
+    /**
+     * The stage of the virtual network definition which contains all the minimum required inputs for
+     * the resource to be created (via {@link DefinitionCreatable#create()}), but also allows 
+     * for any other optional settings to be specified, except for adding subnets.
+     * <p>
+     * Subnets can be added only right after the address space is explicitly specified 
+     * (see {@link DefinitionWithAddressSpace#withAddressSpace(String)).
+     */
+    interface DefinitionCreatable extends 
+        Creatable<Network>,
+        Resource.DefinitionWithTags<DefinitionCreatable> { 
+        
+        /**
+         * Specifies the IP address of an existing DNS server to associate with the virtual network.
+         * @param ipAddress the IP address of the DNS server
+         * @return the next stage of the virtual network definition
+         */
+        DefinitionCreatable withDnsServer(String ipAddress);
+
+        /**
+         * Explicitly adds an address space to the virtual network.
+         * <p>
+         * If no address spaces are explicitly specified, a default address space with the CIDR "10.0.0.0/16" will be
+         * assigned to the virtual network.
+         * <p>
+         * Note that this method's effect is additive, i.e. each time it is used, a new address space is added to the network.
+         * This method does not check for conflicts or overlaps with other address spaces. If there is a conflict, 
+         * a cloud exception may be thrown at the time the network is created.
+         * @param cidr the CIDR representation of the address space
+         * @return the next stage of the virtual network definition
+         */
+        DefinitionCreatableWithSubnet withAddressSpace(String cidr);
+    }
+
+    /**
+     * The stage of the public IP definition which contains all the minimum required inputs for
+     * the resource to be created (via {@link DefinitionCreatable#create()}), but also allows 
+     * for any other optional settings to be specified, including adding subnets.
+     */
+    interface DefinitionCreatableWithSubnet extends
+        DefinitionCreatable,
+        DefinitionWithSubnet {
+    }
+    
+    /**
+     * The template for a public IP address update operation, containing all the settings that 
+     * can be modified.
+     * <p>
+     * Call {@link Update#apply()} to apply the changes to the resource in Azure.
+     */
+    interface Update extends 
+        Appliable<Network>,
+        Resource.UpdateWithTags<Update>,
+        UpdateWithSubnet {
+
+        /**
+         * Specifies the IP address of the DNS server to associate with the virtual network.
+         * @param ipAddress the IP address of the DNS server
+         * @return the next stage of the virtual network update
+         */
+        Update withDnsServer(String ipAddress);
```

Comment:
Having additional information such as "Note this method's effect is additive, i.e. each time it is used, a new dans server is added to the network." will be helpful (similar to withAddressSpace)